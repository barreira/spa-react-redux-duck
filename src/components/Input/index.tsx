import * as React from 'react';

interface InputProps {
  title: string;
  value: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

export const Input: React.FC<InputProps> = ({ title, onChange, value }) => (
  <fieldset>
    <label>
      <div>{title}:</div>
      <input type="text" onChange={onChange} value={value} />
    </label>
  </fieldset>
);

export default Input;
