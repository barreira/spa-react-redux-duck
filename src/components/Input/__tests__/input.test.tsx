import * as React from 'react';
import { mount } from 'enzyme';
import Input from '..';

describe('Input', () => {
  it('Renders Input', () => {
    const input = mount(
      <Input title="Input field" onChange={() => null} value="10" />
    );
    expect(input).toMatchSnapshot();
  });
});
