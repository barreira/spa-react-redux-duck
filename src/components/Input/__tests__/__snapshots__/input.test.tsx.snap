// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Renders Input 1`] = `ReactWrapper {}`;
